File: challs/command_injection.dir/command_injection.go
package main

import (
    "fmt"
    "os"
    "regexp"
    "os/exec"
)

// Usage :
// go run ./command_injection.go <server_address>
// ex: go run ./command_injection.go www.ovh.com
// do a dig on this server

func main() {
    // check params
    args := os.Args[1:]
    if len(args) != 1 || args[0] == "" {
        fmt.Println("Please send me a server address so I can dig it ! (ex: google.com or www.ovh.com)")
        os.Exit(0)
    }
    server := args[0]
    regexServerCompiled := regexp.MustCompile("[!;&|'\"`${}><]")
    if regexServerCompiled.MatchString(server){
        fmt.Println("Error, server param not valid.")
        os.Exit(0)
    }
    // end check params

    // launch dig
    command := "/usr/bin/dig +short " + server
    resultDig, _ := exec.Command("sh", "-c", command).CombinedOutput()
    // end launch dig

    fmt.Println(string(resultDig[:]))
}

File: challs/command_injection.dir/command_injection.pl
#!/usr/bin/perl -w
use warnings;
use strict;

## Usage :
# ./command_injection.pl <server_address>
# ex: ./command_injection.pl www.ovh.com
# do a dig on this server

## check params
if (@ARGV == 0 || !$ARGV[0]) {
	print "Please send me a server address so I can dig it ! (ex: google.com or www.ovh.com)\n";
	exit 0;
}
my $server = $ARGV[0];
if ($server =~ /[!;&\|'"`\${}><]/){
    print "Error, server param not valid.\n";
    exit 0;
}
## end check params

## launch dig
my $resultDig = `/usr/bin/dig +short $server`;
## end launch dig

print "$resultDig\n";

1;
